src/statscollector.h

```diff
@@ -0,0 +1,215 @@
+		unsigned int nConnectionCount;
+  
+	};
+
+	class ConnectionMinerData
+	{
+		public:
+		ConnectionMinerData(){PPS = 0, WPS = 0, GUID="";}
+		ConnectionMinerData(std::string lGUID){GUID = lGUID, PPS = 0, WPS = 0;}
+
+		double PPS, WPS;
+		std::string GUID;
+	};
+
+	class ConnectionData
+	{
+		public:
+		ConnectionData(){ADDRESS="";}
+		ConnectionData(std::string lAddress){ADDRESS = lAddress, LASTSAVETIMER.Start();}
+
+		double GetTotalPPS() const 
+		{
+			double PPS = 0;
+			for(std::map<std::string, ConnectionMinerData >::const_iterator iter = MINER_DATA.begin(); iter != MINER_DATA.end(); ++iter)
+			{
+				PPS += iter->second.PPS;
+			}
+
+			return PPS;
+		}
+		double GetTotalWPS() const 
+		{
+			double WPS = 0;
+			for(std::map<std::string, ConnectionMinerData >::const_iterator iter = MINER_DATA.begin(); iter != MINER_DATA.end(); ++iter)
+			{
+				WPS += iter->second.WPS;
+			}
+
+			return WPS;
+		}
+
+		std::string ADDRESS;
+		std::map<std::string, ConnectionMinerData > MINER_DATA;
+
+		/** Used to track when this connection data was last persisted**/
+		LLP::Timer LASTSAVETIMER;
+	};
+
+	class RoundData
+	{
+	public:
+		RoundData(int nRoundToSet)
+		{
+			nRound = nRoundToSet;
+		}
+
+		RoundData()
+		{
+			nRound = Core::nCurrentRound;
+			nBlockNumber = Core::nBestHeight;
+			hashBlock = Core::hashBlockSubmission.ToString(); 
+			nRoundReward = Core::nRoundReward;
+			nTotalShares = Core::TotalWeight();
+			strBlockFinder = Core::LAST_ROUND_BLOCKFINDER;
+			bOrphan = false;
+			tBlockFoundTime = time(0);
+		}
+
+
+		unsigned int nRound;
+		unsigned int nBlockNumber;
+        std::string hashBlock;
+        uint64 nRoundReward;
+		uint64 nTotalShares;
+		std::string strBlockFinder;
+		bool bOrphan;
+        time_t tBlockFoundTime;
+	};
+
+	class AccountData
+	{
+	public:
+
+		std::string strAccountAddress;
+        int nConnections;
+		uint64 nShares;
+        uint64_t nBalance ;
+        uint64_t nPendingPayout;
+
+	};
+
+	class AccountEarningTransaction
+	{
+	public:
+		AccountEarningTransaction(){}
+
+		std::string strAccountAddress;
+		unsigned int nRound;
+		unsigned int nBlockNumber;
+		uint64 nShares;
+		uint64 nAmountEarned;
+		time_t tTime;
+
+	};
+
+	class AccountPaymentTransaction
+	{	
+	public:
+		AccountPaymentTransaction(){};
+
+		std::string strAccountAddress;
+		unsigned int nRound;
+		unsigned int nBlockNumber;
+		unsigned int nAmountPaid;
+		time_t tTime;
+
+	};
+
+	/** Class used to generate JSON-formatted pool statistics **/
+	class StatsCollector
+	{
+	public:
+
+		StatsCollector(){STATSPERSISTER = NULL; }
+		~StatsCollector();
+		void Init();
+
+
+		void IncConnectionCount( std::string ADDRESS, std::string GUID );
+		void DecConnectionCount( std::string ADDRESS, std::string GUID );
+		int GetConnectionCount( std::string ADDRESS );
+		void UpdateConnectionData(std::string ADDRESS, std::string GUID, double PPS, double WPS);
+		double GetAccountPPS(std::string ADDRESS);
+		double GetAccountWPS(std::string ADDRESS);
```

Comment:
I think it's better to remove or comment out variables and methods that aren't used - otherwise people have to maintain them and/or wonder what they are doing there.